sympy/polys/matrices/domainmatrix.py

```diff
@@ -242,6 +241,18 @@ def from_Matrix(cls, M, **kwargs):
         Matrix
 
         """
+        m, n = M.shape
+        non_zero = 0
+
+        for i in range(m):
+            for j in range(n):
+                if not M[i, j].is_zero:
```

Comment:
`is_zero` can return `None`.

Reply:
Calling `is_zero` here is potentially slow for some expressions. I suggest just checking for an exact zero like `M[i, i] is S.Zero`.